content/doc/book/platform-information/upgrade-java-to-21.adoc

```diff
@@ -0,0 +1,142 @@
+---
+layout: subsection
+title: Upgrade to Java 21
+---
+
+When upgrading the JVM used to run Jenkins from Java 17 to Java 21, there are some details you should know and precautions you should take.
+NOTE: Java 21 is supported as of LTS 2.426.1 and Jenkins Weekly 2.419.
+
+== Back up Jenkins
+
+As with any upgrade, we recommend:
+
+. link:https:/doc/book/system-administration/backing-up/#jenkins_home[Backing up `JENKINS_HOME`].
+. Testing the upgrade with your backup.
+. After all required tests pass, execute the upgrade on your production controller.
+
+=== Upgrade the Jenkins Java version to Java 21
+
+.Upgrading Jenkins Java Version From 17 to 21
+video::8xQVGpWeIe0[youtube,width=800,height=420]
+
+To verify the Java version currently used in your controller:
+
+. Navigate to *Manage Jenkins* and then select *System Information* in the *Status Information* section.
+. On the *System Properties* tab, locate the `java.runtime.version` and reveal the hidden value to display your current Java version.
+
+To verify the Java version currently used in your agent:
+
+. Select an agent name from the *Build Executor Status* widget and then select *System Information*.
+. Locate the `java.runtime.version` and reveal the hidden value to display the current Java version.
+
+The `checkNodes` script determines what version of Java is running the controller process, along with the version of the agent that's associated with that controller.
+The script then checks the version of Java that is on the agent.
+When these values are returned, `OK` means that the agent Java version matches the controller Java version.
+Otherwise, the result displays the expected Java version and the version found instead.
+
+7To run the `checkNodes` script:
```

Comment:
```suggestion
To run the `checkNodes` script:
```

Reply:
Those darn cats are trying to sabotage me.